[Ai4dSoc][Serverless] Hide `Security` feature sub-privileges in `search_ai_lake` tier

ashokaditya opened the pull request:
## Summary

Hides security sub-privileges for ai4soc/search_ai_lake tier.
![Screenshot 2025-04-11 at 10 22 09](https://github.com/user-attachments/assets/6f3294bc-82de-404e-b9d3-22e717d54b65)

### Reasoning for changes added to `x-pack/packages/security`:

Currently, the feature description of Security feature is tied to the fact that it has a list of sub-privileges. This is true on ESS and `essentials/complete` serverless tiers.

With the introduction of the lower `search_ai_lake` tier, security feature would not have any sub-privileges available and thus it does not make sense to show that description. 

The ideal way to handle this would be to load feature privileges config settings at the plugin level (security_solution/security_solution_serverless) and set `description` to `null | undefined` based on the tier, as currently the feature privileges settings live in [kibana_features file (v2_features)](https://github.com/elastic/kibana/blob/795094d8c63dde1439220cfba1808fce50c781d1/x-pack/solutions/security/packages/features/src/security/v2_features/kibana_features.ts#L72) (also another set in v1_features) and the plugins only select a set of those based on the [feature keys available](https://github.com/elastic/kibana/blob/d4a33a2b61f05e85f99f8f5932b0266babb439f2/x-pack/solutions/security/plugins/security_solution_serverless/common/pli/pli_config.ts) on each tier. The refactoring to pass in feature configs at the plugin level (instead of just feature keys) is not in the scope of the work cut out for RSA conf.

Thus the other simpler approach in this PR is to allow overriding the description field on the tier specific config file.

## How to Test

1. While on the Kibana root directory, run ES/Kibana on serverless mode with:

```bash
yarn es serverless --kill --projectType security --kibanaUrl=http://0.0.0.0:5601
```
and on a new window
```bash
yarn serverless-security --no-base-path
```

Enable the AI for SOC tier, by adding the following to your `serverless.security.dev.yaml` file:

```json5
xpack.securitySolutionServerless.productTypes:
  [
    { product_line: 'ai_soc', product_tier: 'search_ai_lake' },
  ]
```

2. Once Kibana is up and running login in with the `admin` role using the role dropdown.
3. Navigate to `app/management/roles/edit`
4. Click on `Assign to space` button and assign a space to that role on the `Assign role to spaces` flyout.
5. Expand the `Security` category and verify that `Security` feature is listed in the list of features.
6. Also verify that there is neither an accordion icon beside `Security` feature nor a description text under it about sub-privileges.

### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...





<!--ONMERGE {"backportTargets":["8.19"]} ONMERGE-->

ashokaditya committed:
hide security sub-privileges in `search_ai_lake` tier

closes https://github.com/elastic/security-team/issues/12045

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

ashokaditya committed:
tests

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/290632)
* Commit: 693d503a35d6e440841bdbfd5e964f9ab9bbb587

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/645353049787a0b2a07f67be108f5c70942a58f2) | [after](https://github.com/elastic/kibana/commit/693d503a35d6e440841bdbfd5e964f9ab9bbb587) | diff |
| --- | --- | --- | --- |
| `securitySolutionServerless` | 51.3KB | 51.2KB | -70.0B |



cc @ashokaditya

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290632","number":290632,"commit":"693d503a35d6e440841bdbfd5e964f9ab9bbb587"}],"number":290632}
buildkite-pr-comment-->

elasticmachine commented:
Pinging @elastic/security-defend-workflows (Team:Defend Workflows)

semd commented on x-pack/test/security_solution_cypress/ai4dsoc_serverless_config.ts:
```diff
@@ -34,6 +34,7 @@ export default async function ({ readConfigFile }: FtrConfigProviderContext) {
         ])}`,
         '--csp.strict=false',
         '--csp.warnLegacyBrowsers=false',
+        '--xpack.features.overrides.siemV2.hidden=true',
```
Why do we need to do this?

semd commented on x-pack/test/security_solution_cypress/ai4dsoc_serverless_config.ts:
```diff
@@ -34,6 +34,7 @@ export default async function ({ readConfigFile }: FtrConfigProviderContext) {
         '--csp.strict=false',
         '--csp.warnLegacyBrowsers=false',
+        '--xpack.features.overrides.siemV2.hidden=true',
         '--xpack.features.overrides.securitySolutionTimeline.hidden=true',
         '--xpack.features.overrides.securitySolutionNotes.hidden=true',
```
Also, I still don't get why we need to duplicate these overrides:
```ts
        '--xpack.features.overrides.securitySolutionTimeline.hidden=true',
        '--xpack.features.overrides.securitySolutionNotes.hidden=true',
```
The test server should already have that setup from the tier config (`config/serverless.security.search_ai_lake.yml`)

semd commented on config/serverless.security.search_ai_lake.yml:
```diff
@@ -7,8 +7,9 @@ xpack.features.overrides:
   ### The following features are Security features hidden in Role management UI for this specific tier.
   securitySolutionTimeline.hidden: true
   securitySolutionNotes.hidden: true
+  siemV2.hidden: true
```
Why do we need to hide this?

ashokaditya replied:
It is so that we hide `security` sub privileges in the roles UI for ai4soc. Correct?

semd replied:
This hides the entire _Security_ feature. Is that intentional?

ashokaditya replied:
No it doesn't hide the entire Security feature (as you see in the screenshot) it hides only the security sub-feature that has privileges such as host isolation, response actions, trusted apps etc.

Here's the [parent issue](https://github.com/elastic/security-team/issues/11781) and a [breakdown issue ](https://github.com/elastic/security-team/issues/12040)

semd replied:
I see, you want to hide the features:
<img width="754" alt="Captura de pantalla 2025-04-09 a les 10 29 02" src="https://github.com/user-attachments/assets/be3bdaf5-524f-4186-bf09-6a4c4a36386c" />
<img width="754" alt="Captura de pantalla 2025-04-09 a les 10 30 47" src="https://github.com/user-attachments/assets/14ff101b-6939-43f3-90ce-0cf1b9858a4b" />

semd replied:
https://github.com/elastic/security-team/issues/11781#issuecomment-2788825208

ashokaditya replied:
Oh I see, it's a "feature" like you call it and not sub-feature. 

I actually see it as Security "feature" (not category)  with a list of sub-features within it, one of which is a Security "sub-feature" with list of "privileges". Same with Management feature and a list of privileges listed under it.

But either way the idea is to hide the Security "feature" (sub-feature as I call it).

ashokaditya replied:
Oh okay I see what you mean with the alerts not working as the Security feature controls other areas of security as well. I'll then go ahead and try and hide the "sub-feature" privilege (same as in your screenshots)

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

ashokaditya committed:
review changes

ashokaditya committed:
add a hostIsolation feature key

This is to allow adding the sub feature privilege to specific tiers, similar to response actions, endpoint artifacts etc. This change is needed so that we can restrict all security sub-feature privileges in the search_ai_lake tier.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/291958)
* Commit: 3b5e172de15b5a19f78ad1504ce1011555bd47e3

### Failed CI Steps
* [Serverless AI4DSOC - Security Solution Cypress Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/291958#01961fb3-0781-456d-ba41-eae60b509309)

### History
* :green_heart: [Build #290632](https://buildkite.com/elastic/kibana-pull-request/builds/290632) succeeded 693d503a35d6e440841bdbfd5e964f9ab9bbb587

cc @ashokaditya

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291958","number":291958,"commit":"3b5e172de15b5a19f78ad1504ce1011555bd47e3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290632","number":290632,"commit":"693d503a35d6e440841bdbfd5e964f9ab9bbb587"}],"number":291958}
buildkite-pr-comment-->

ashokaditya committed:
update tests

ashokaditya committed:
Conditionally show sub-privilege description

No need to show sub-privilege description (if any) when no sub-privileges under a feature.

ashokaditya commented on x-pack/platform/packages/private/security/ui_components/src/kibana_privilege_table/components/feature_table_cell.tsx:
```diff
@@ -37,6 +37,14 @@ export const FeatureTableCell = ({ feature, className }: Props) => {
+    () =>
+      feature.description &&
+      feature.description.length > 0 &&
+      feature.getSubFeaturePrivileges().length > 0,
```
This logic is to hide description for feature added in https://github.com/elastic/kibana/pull/147709 and updated in https://github.com/elastic/kibana/pull/192273

ashokaditya replied:
You are right @azasypkin. I've reverted the change.

azasypkin replied:
Great, thanks 👍

ashokaditya committed:
Merge branch 'main' into task/ai4soc-hide-security-sub-privileges

ashokaditya committed:
update tests

ashokaditya committed:
remove duplicate config flags

ashokaditya committed:
Merge branch 'main' into task/ai4soc-hide-security-sub-privileges

tomsonpl reviewed (approved):
Tested locally, looks good 👍

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/292237)
* Commit: 9610e96f2b5e435a9a8eb9d2b85aa0ab8e696085

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/cb5ae3a48359e35461fe804aacbe5a6af1fe360e) | [after](https://github.com/elastic/kibana/commit/9610e96f2b5e435a9a8eb9d2b85aa0ab8e696085) | diff |
| --- | --- | --- | --- |
| `security` | 515.6KB | 515.8KB | +159.0B |
| `spaces` | 209.2KB | 209.4KB | +157.0B |
| total |  |  | +316.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/cb5ae3a48359e35461fe804aacbe5a6af1fe360e) | [after](https://github.com/elastic/kibana/commit/9610e96f2b5e435a9a8eb9d2b85aa0ab8e696085) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 88.5KB | 88.6KB | +50.0B |
| `securitySolutionEss` | 33.0KB | 33.1KB | +50.0B |
| `securitySolutionServerless` | 47.8KB | 47.8KB | +88.0B |
| total |  |  | +188.0B |



### History
* :broken_heart: [Build #291958](https://buildkite.com/elastic/kibana-pull-request/builds/291958) failed 3b5e172de15b5a19f78ad1504ce1011555bd47e3
* :green_heart: [Build #290632](https://buildkite.com/elastic/kibana-pull-request/builds/290632) succeeded 693d503a35d6e440841bdbfd5e964f9ab9bbb587

cc @ashokaditya

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292237","number":292237,"commit":"9610e96f2b5e435a9a8eb9d2b85aa0ab8e696085"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291958","number":291958,"commit":"3b5e172de15b5a19f78ad1504ce1011555bd47e3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290632","number":290632,"commit":"693d503a35d6e440841bdbfd5e964f9ab9bbb587"}],"number":292237}
buildkite-pr-comment-->

semd commented on x-pack/solutions/security/plugins/security_solution_serverless/common/pli/pli_config.ts:
```diff
@@ -59,12 +62,14 @@ export const PLI_PRODUCT_FEATURES: PliProductFeatures = {
       ProductFeatureKey.endpointPolicyProtections,
       ProductFeatureKey.endpointArtifactManagement,
       ProductFeatureKey.endpointExceptions,
+      ProductFeatureKey.endpointHostIsolation,
     ],
     complete: [
       ProductFeatureKey.endpointPolicyProtections,
       ProductFeatureKey.endpointArtifactManagement,
       ProductFeatureKey.endpointExceptions,
       ProductFeatureKey.endpointHostIsolationExceptions,
+      ProductFeatureKey.endpointHostIsolation,
```
Why are they duplicated in the `ProductLine.endpoint` add-on? They are already included in the `ProductLine.security` base PLI.

ashokaditya replied:
removed the duplicates

semd commented on x-pack/solutions/security/plugins/security_solution_serverless/common/pli/pli_config.ts:
```diff
@@ -22,8 +22,9 @@ export const PLI_PRODUCT_FEATURES: PliProductFeatures = {
     ],
-    essentials: [ProductFeatureKey.attackDiscovery, ProductFeatureKey.assistant],
-    complete: [ProductFeatureKey.attackDiscovery, ProductFeatureKey.assistant],
+    // neither of these tiers are available in ai_soc product line
+    essentials: [],
+    complete: [],
```
Actually, the `ProductLine.aiSoc` add-on will be available for _essentials_ (not sure _complete_) eventually, right?

ashokaditya replied:
Maybe, but from the UX and discussions so far it looks like ai4Soc does not have those tiers at all.

semd replied:
I thought the plan was to, eventually, be able to add the `ai4Soc` add-on to the _essentials_ tier, to add AI features to it. Has the plan changed?

semd replied:
Maybe you are confusing `ai4Soc` add-on with the `searchAiLake` tier?

tomsonpl replied:
@semd you're right 👍 `ai_soc` will be available in `essentials/complete` in the future.

This comment sums up the current and the next phases: https://github.com/elastic/security-team/issues/11781#issuecomment-2701518004

ashokaditya committed:
Update pli_config.ts

semd commented on x-pack/solutions/security/packages/features/src/security/product_feature_config.ts:
```diff
@@ -111,6 +111,11 @@ export const securityDefaultProductFeaturesConfig: DefaultSecurityProductFeature
     },
   },
 
+  // Endpoint Essentials/Complete tiers
```
This file should be tier-agnostic; this information is already in the `pli_config.ts`

ashokaditya replied:
Got it! I'll remove the comment

ashokaditya committed:
Update product_feature_config.ts

vgomez-el reviewed (approved):
LGTM!

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/292371)
* Commit: 019eca25334f171cf775dea89311d62e45c1b57f

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/cb5ae3a48359e35461fe804aacbe5a6af1fe360e) | [after](https://github.com/elastic/kibana/commit/019eca25334f171cf775dea89311d62e45c1b57f) | diff |
| --- | --- | --- | --- |
| `security` | 515.6KB | 515.8KB | +159.0B |
| `spaces` | 209.2KB | 209.4KB | +157.0B |
| total |  |  | +316.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/cb5ae3a48359e35461fe804aacbe5a6af1fe360e) | [after](https://github.com/elastic/kibana/commit/019eca25334f171cf775dea89311d62e45c1b57f) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 88.5KB | 88.6KB | +50.0B |
| `securitySolutionEss` | 33.0KB | 33.1KB | +50.0B |
| `securitySolutionServerless` | 47.8KB | 47.8KB | +34.0B |
| total |  |  | +134.0B |



### History
* :green_heart: [Build #292237](https://buildkite.com/elastic/kibana-pull-request/builds/292237) succeeded 9610e96f2b5e435a9a8eb9d2b85aa0ab8e696085
* :broken_heart: [Build #291958](https://buildkite.com/elastic/kibana-pull-request/builds/291958) failed 3b5e172de15b5a19f78ad1504ce1011555bd47e3
* :green_heart: [Build #290632](https://buildkite.com/elastic/kibana-pull-request/builds/290632) succeeded 693d503a35d6e440841bdbfd5e964f9ab9bbb587

cc @ashokaditya

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292371","number":292371,"commit":"019eca25334f171cf775dea89311d62e45c1b57f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292237","number":292237,"commit":"9610e96f2b5e435a9a8eb9d2b85aa0ab8e696085"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291958","number":291958,"commit":"3b5e172de15b5a19f78ad1504ce1011555bd47e3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290632","number":290632,"commit":"693d503a35d6e440841bdbfd5e964f9ab9bbb587"}],"number":292371}
buildkite-pr-comment-->

ashokaditya committed:
Merge branch 'main' into task/ai4soc-hide-security-sub-privileges

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/292521)
* Commit: 2ab64914ef31231cd6df8f0156cf54827e51a32d

### Failed CI Steps
* [Build Kibana Distribution](https://buildkite.com/elastic/kibana-pull-request/builds/292521#01963327-f776-4ec4-8a52-e12ff056d098)

### History
* :green_heart: [Build #292371](https://buildkite.com/elastic/kibana-pull-request/builds/292371) succeeded 019eca25334f171cf775dea89311d62e45c1b57f
* :green_heart: [Build #292237](https://buildkite.com/elastic/kibana-pull-request/builds/292237) succeeded 9610e96f2b5e435a9a8eb9d2b85aa0ab8e696085
* :broken_heart: [Build #291958](https://buildkite.com/elastic/kibana-pull-request/builds/291958) failed 3b5e172de15b5a19f78ad1504ce1011555bd47e3
* :green_heart: [Build #290632](https://buildkite.com/elastic/kibana-pull-request/builds/290632) succeeded 693d503a35d6e440841bdbfd5e964f9ab9bbb587

cc @ashokaditya

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292521","number":292521,"commit":"2ab64914ef31231cd6df8f0156cf54827e51a32d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292371","number":292371,"commit":"019eca25334f171cf775dea89311d62e45c1b57f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292237","number":292237,"commit":"9610e96f2b5e435a9a8eb9d2b85aa0ab8e696085"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291958","number":291958,"commit":"3b5e172de15b5a19f78ad1504ce1011555bd47e3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290632","number":290632,"commit":"693d503a35d6e440841bdbfd5e964f9ab9bbb587"}],"number":292521}
buildkite-pr-comment-->

ashokaditya commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/292551)
* Commit: 2ab64914ef31231cd6df8f0156cf54827e51a32d

### Failed CI Steps
* [Build Kibana Distribution](https://buildkite.com/elastic/kibana-pull-request/builds/292551#01963354-ee0f-4d40-b983-aa19fa20596a)

### History
* :broken_heart: [Build #292521](https://buildkite.com/elastic/kibana-pull-request/builds/292521) failed 2ab64914ef31231cd6df8f0156cf54827e51a32d
* :green_heart: [Build #292371](https://buildkite.com/elastic/kibana-pull-request/builds/292371) succeeded 019eca25334f171cf775dea89311d62e45c1b57f
* :green_heart: [Build #292237](https://buildkite.com/elastic/kibana-pull-request/builds/292237) succeeded 9610e96f2b5e435a9a8eb9d2b85aa0ab8e696085
* :broken_heart: [Build #291958](https://buildkite.com/elastic/kibana-pull-request/builds/291958) failed 3b5e172de15b5a19f78ad1504ce1011555bd47e3
* :green_heart: [Build #290632](https://buildkite.com/elastic/kibana-pull-request/builds/290632) succeeded 693d503a35d6e440841bdbfd5e964f9ab9bbb587

cc @ashokaditya

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292551","number":292551,"commit":"2ab64914ef31231cd6df8f0156cf54827e51a32d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292521","number":292521,"commit":"2ab64914ef31231cd6df8f0156cf54827e51a32d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292371","number":292371,"commit":"019eca25334f171cf775dea89311d62e45c1b57f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292237","number":292237,"commit":"9610e96f2b5e435a9a8eb9d2b85aa0ab8e696085"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/291958","number":291958,"commit":"3b5e172de15b5a19f78ad1504ce1011555bd47e3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/290632","number":290632,"commit":"693d503a35d6e440841bdbfd5e964f9ab9bbb587"}],"number":292551}
buildkite-pr-comment-->

semd reviewed (approved):
LGTM! thanks @ashokaditya

ashokaditya committed:
Merge branch 'main' into task/ai4soc-hide-security-sub-privileges

azasypkin reviewed (changes requested):
The way of hiding the feature description in the feature table feels wrong to me, we should find another way.

ashokaditya committed:
Revert "Conditionally show sub-privilege description"

This reverts commit 5d86882e178bd43930750e98674a607483b63535.

ashokaditya committed:
hide security feature description (no-sub-privi)

This change is ensure we do not show the Security feature description when all sub-privileges are hidden (on one or several specific tiers)

ashokaditya committed:
Revert "hide security feature description (no-sub-privi)"

This reverts commit aa1e60a0132abaff0430cd809d145ed3f7a2fb6b.

ashokaditya committed:
override feature description instead

ashokaditya committed:
Update config.test.ts

ashokaditya committed:
Merge branch 'main' into task/ai4soc-hide-security-sub-privileges

ashokaditya committed:
nullable description

ashokaditya committed:
Update feature_registry.ts

kc13greiner commented on x-pack/platform/plugins/shared/features/server/feature_registry.ts:
```diff
@@ -123,6 +123,12 @@ export class FeatureRegistry {
 
+      // We are allowing null values for description to be able to remove the description from the feature.
+      // The override is used in serverless.security.search_ai_lake.yml
+      if (typeof featureOverride.description !== 'undefined') {
```
Question: What is the reason for doing this so far downstream? Could this sort of logic be handled in the feature registration of the plugin that it is required for?

ashokaditya replied:
@kc13greiner Thanks for the review. Currently at the [plugin level](https://github.com/elastic/kibana/blob/44f072bbe5c5edc6ebf2ab50f0e70bbd4a47e7fd/x-pack/solutions/security/plugins/security_solution_serverless/server/plugin.ts#L79) we select the feature keys to show and that sets the [feature config from the security package](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/packages/features/src/security/product_feature_config.ts).  If we want to set description of the features at the plugin level then we need to pass each product feature config at the plugin level. So yes it can be done at plugin setup stage, but it would need a lot of code changes/refactoring to sort it out.

For now this approach of being able to override the feature description seems the best and simplest (similar to `name` property) given the scope of the work cut out for Apr 28th.

cc @semd @YulNaumenko Let me know if you think otherwise or I got anything incorrect above.

rudolf replied:
I'm concerned about the complexity of introducing all these config level overrides and the disconnect it introduces with two sources of truth: 1) the plugin API calls and 2) different product tier configuration files

Has there been any discussions around the level of effort it would require to register feature descriptions and hidden attributes programatically? It sounds like the only reason this approach has been rejected is due to time constraints.

rudolf replied:
Approving for now, follow-up refactoring work will be done after Apr 28 and tracked in https://github.com/elastic/kibana/issues/218463

kc13greiner commented on x-pack/platform/plugins/shared/features/server/config.test.ts:
```diff
@@ -39,7 +39,7 @@ describe('config schema', () => {
         {
           overrides: {
-            featureA: { name: 'new name', hidden: true },
+            featureA: { name: 'new name', description: 'new description', hidden: true },
```
If this logic does stay in this plugin, tests should be added for the new `null` case

ashokaditya replied:
done

kc13greiner reviewed (commented):
Heya! Just a couple questions. Please see below!

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/292831)
* Commit: 020c0224edc3039635de7b2d61e7c6fb1ed4a695

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/6722f142a4d36f0c84c9eb258287900f2f559389) | [after](https://github.com/elastic/kibana/commit/020c0224edc3039635de7b2d61e7c6fb1ed4a695) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 88.8KB | 88.8KB | +50.0B |
| `securitySolutionEss` | 33.5KB | 33.6KB | +50.0B |
| `securitySolutionServerless` | 48.3KB | 48.3KB | +34.0B |
| total |  |  | +134.0B |



### History
* :broken_heart: [Build #292596](https://buildkite.com/elastic/kibana-pull-request/builds/292596) failed 416a928e3f440f3c2d44aae4ceb7a4864ca516ad
* :broken_heart: [Build #292551](https://buildkite.com/elastic/kibana-pull-request/builds/292551) failed 2ab64914ef31231cd6df8f0156cf54827e51a32d
* :broken_heart: [Build #292521](https://buildkite.com/elastic/kibana-pull-request/builds/292521) failed 2ab64914ef31231cd6df8f0156cf54827e51a32d
* :green_heart: [Build #292371](https://buildkite.com/elastic/kibana-pull-request/builds/292371) succeeded 019eca25334f171cf775dea89311d62e45c1b57f
* :green_heart: [Build #292237](https://buildkite.com/elastic/kibana-pull-request/builds/292237) succeeded 9610e96f2b5e435a9a8eb9d2b85aa0ab8e696085
* :broken_heart: [Build #291958](https://buildkite.com/elastic/kibana-pull-request/builds/291958) failed 3b5e172de15b5a19f78ad1504ce1011555bd47e3

cc @ashokaditya

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292831","number":292831,"commit":"020c0224edc3039635de7b2d61e7c6fb1ed4a695"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292596","number":292596,"commit":"416a928e3f440f3c2d44aae4ceb7a4864ca516ad"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292551","number":292551,"commit":"2ab64914ef31231cd6df8f0156cf54827e51a32d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292521","number":292521,"commit":"2ab64914ef31231cd6df8f0156cf54827e51a32d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292371","number":292371,"commit":"019eca25334f171cf775dea89311d62e45c1b57f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292237","number":292237,"commit":"9610e96f2b5e435a9a8eb9d2b85aa0ab8e696085"}],"number":292831}
buildkite-pr-comment-->

ashokaditya committed:
Update config.test.ts

ashokaditya committed:
Merge branch 'main' into task/ai4soc-hide-security-sub-privileges

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/293251)
* Commit: 64099c7e6f65d0beef03fd943b35d85b5098b86f

### Failed CI Steps
* [FTR Configs #13](https://buildkite.com/elastic/kibana-pull-request/builds/293251#019639cd-752c-47bc-aab5-847614cdb288)
* [FTR Configs #50](https://buildkite.com/elastic/kibana-pull-request/builds/293251#019639cd-7540-4d2a-b180-8c43d0f3b57b)
* [FTR Configs #50](https://buildkite.com/elastic/kibana-pull-request/builds/293251#019639f4-278e-4c4a-90c3-bd45482f19d8)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/293251#019639cd-752c-47bc-aab5-847614cdb288) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/293251/jobs/019639cd-752c-47bc-aab5-847614cdb288/artifacts/019639fa-d16b-4294-9761-70c09b85854e) FTR Configs #<span></span>13 / @<span></span>ess @<span></span>serverless SecuritySolution Explore Overview Overview Host With auditbeat Make sure that we get OverviewHost data
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/293251#019639cd-7540-4d2a-b180-8c43d0f3b57b) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/293251/jobs/019639cd-7540-4d2a-b180-8c43d0f3b57b/artifacts/019639f4-04c8-42fa-bed5-7400750df0cf) FTR Configs #<span></span>50 / console app misc console behavior keyboard shortcuts open documentation should open documentation when Ctrl+/ is pressed
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/293251#019639f4-278e-4c4a-90c3-bd45482f19d8) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/293251/jobs/019639f4-278e-4c4a-90c3-bd45482f19d8/artifacts/019639ff-ad1f-41d3-be81-6aa285afb756) FTR Configs #<span></span>50 / console app misc console behavior keyboard shortcuts open documentation should open documentation when Ctrl+/ is pressed

### History
* :green_heart: [Build #292831](https://buildkite.com/elastic/kibana-pull-request/builds/292831) succeeded 020c0224edc3039635de7b2d61e7c6fb1ed4a695
* :broken_heart: [Build #292596](https://buildkite.com/elastic/kibana-pull-request/builds/292596) failed 416a928e3f440f3c2d44aae4ceb7a4864ca516ad
* :broken_heart: [Build #292551](https://buildkite.com/elastic/kibana-pull-request/builds/292551) failed 2ab64914ef31231cd6df8f0156cf54827e51a32d
* :broken_heart: [Build #292521](https://buildkite.com/elastic/kibana-pull-request/builds/292521) failed 2ab64914ef31231cd6df8f0156cf54827e51a32d
* :green_heart: [Build #292371](https://buildkite.com/elastic/kibana-pull-request/builds/292371) succeeded 019eca25334f171cf775dea89311d62e45c1b57f
* :green_heart: [Build #292237](https://buildkite.com/elastic/kibana-pull-request/builds/292237) succeeded 9610e96f2b5e435a9a8eb9d2b85aa0ab8e696085

cc @ashokaditya

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293251","number":293251,"commit":"64099c7e6f65d0beef03fd943b35d85b5098b86f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292831","number":292831,"commit":"020c0224edc3039635de7b2d61e7c6fb1ed4a695"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292596","number":292596,"commit":"416a928e3f440f3c2d44aae4ceb7a4864ca516ad"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292551","number":292551,"commit":"2ab64914ef31231cd6df8f0156cf54827e51a32d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292521","number":292521,"commit":"2ab64914ef31231cd6df8f0156cf54827e51a32d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292371","number":292371,"commit":"019eca25334f171cf775dea89311d62e45c1b57f"}],"number":293251}
buildkite-pr-comment-->

ashokaditya committed:
Merge branch 'main' into task/ai4soc-hide-security-sub-privileges

kc13greiner commented on x-pack/platform/plugins/shared/features/server/feature_registry.ts:
```diff
@@ -123,6 +123,12 @@ export class FeatureRegistry {
       }
 
+      // We are allowing null values for description to be able to remove the description from the feature.
+      // The override is used in serverless.security.search_ai_lake.yml
```
Nit: Please remove this line.

kc13greiner replied:
*to keep the comment about the functionality more generic

ashokaditya replied:
done

kc13greiner reviewed (approved):
Thank you for the changes ( I added 1 additional nit to fix) LGTM!

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/293312)
* Commit: 84412b660f1976eae18f2a481110455395249cfd

### Failed CI Steps
* [FTR Configs #47](https://buildkite.com/elastic/kibana-pull-request/builds/293312#01963a26-91c3-4148-822e-3b544e80eb38)
* [FTR Configs #48](https://buildkite.com/elastic/kibana-pull-request/builds/293312#01963a26-91c3-4837-94c6-762881eee38f)
* [FTR Configs #48](https://buildkite.com/elastic/kibana-pull-request/builds/293312#01963a49-95fa-4348-94b4-5d1125d67cee)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/293312#01963a26-91c3-4148-822e-3b544e80eb38) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/293312/jobs/01963a26-91c3-4148-822e-3b544e80eb38/artifacts/01963a4f-3a50-4ba8-ac11-7d268810df6e) FTR Configs #<span></span>47 / @<span></span>ess @<span></span>serverless SecuritySolution Explore Overview Overview Host With auditbeat Make sure that we get OverviewHost data
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/293312#01963a26-91c3-4837-94c6-762881eee38f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/293312/jobs/01963a26-91c3-4837-94c6-762881eee38f/artifacts/01963a49-7d97-4ad8-b182-d3b84fe9a084) FTR Configs #<span></span>48 / console app console context menu should open documentation when open documentation button is clicked
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/293312#01963a49-95fa-4348-94b4-5d1125d67cee) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/293312/jobs/01963a49-95fa-4348-94b4-5d1125d67cee/artifacts/01963a54-fe04-4675-bb35-2990d564cf5f) FTR Configs #<span></span>48 / console app console context menu should open documentation when open documentation button is clicked

### History
* :broken_heart: [Build #293251](https://buildkite.com/elastic/kibana-pull-request/builds/293251) failed 64099c7e6f65d0beef03fd943b35d85b5098b86f
* :green_heart: [Build #292831](https://buildkite.com/elastic/kibana-pull-request/builds/292831) succeeded 020c0224edc3039635de7b2d61e7c6fb1ed4a695
* :broken_heart: [Build #292596](https://buildkite.com/elastic/kibana-pull-request/builds/292596) failed 416a928e3f440f3c2d44aae4ceb7a4864ca516ad
* :broken_heart: [Build #292551](https://buildkite.com/elastic/kibana-pull-request/builds/292551) failed 2ab64914ef31231cd6df8f0156cf54827e51a32d
* :broken_heart: [Build #292521](https://buildkite.com/elastic/kibana-pull-request/builds/292521) failed 2ab64914ef31231cd6df8f0156cf54827e51a32d
* :green_heart: [Build #292371](https://buildkite.com/elastic/kibana-pull-request/builds/292371) succeeded 019eca25334f171cf775dea89311d62e45c1b57f

cc @ashokaditya

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293312","number":293312,"commit":"84412b660f1976eae18f2a481110455395249cfd"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293251","number":293251,"commit":"64099c7e6f65d0beef03fd943b35d85b5098b86f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292831","number":292831,"commit":"020c0224edc3039635de7b2d61e7c6fb1ed4a695"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292596","number":292596,"commit":"416a928e3f440f3c2d44aae4ceb7a4864ca516ad"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292551","number":292551,"commit":"2ab64914ef31231cd6df8f0156cf54827e51a32d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292521","number":292521,"commit":"2ab64914ef31231cd6df8f0156cf54827e51a32d"}],"number":293312}
buildkite-pr-comment-->

ashokaditya committed:
review changes

ashokaditya committed:
Merge branch 'main' into task/ai4soc-hide-security-sub-privileges

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/293399)
* Commit: d6254594e6cebe8ea27af0226746b46e061c90b0

### Failed CI Steps
* [FTR Configs #119](https://buildkite.com/elastic/kibana-pull-request/builds/293399#01963aad-1d3b-4761-932e-cd76788f83ab)
* [Defend Workflows Cypress Tests #8](https://buildkite.com/elastic/kibana-pull-request/builds/293399#01963a93-eb72-4abb-b5e1-c3081941bef5)

### History
* :broken_heart: [Build #293312](https://buildkite.com/elastic/kibana-pull-request/builds/293312) failed 84412b660f1976eae18f2a481110455395249cfd
* :broken_heart: [Build #293251](https://buildkite.com/elastic/kibana-pull-request/builds/293251) failed 64099c7e6f65d0beef03fd943b35d85b5098b86f
* :green_heart: [Build #292831](https://buildkite.com/elastic/kibana-pull-request/builds/292831) succeeded 020c0224edc3039635de7b2d61e7c6fb1ed4a695
* :broken_heart: [Build #292596](https://buildkite.com/elastic/kibana-pull-request/builds/292596) failed 416a928e3f440f3c2d44aae4ceb7a4864ca516ad
* :broken_heart: [Build #292551](https://buildkite.com/elastic/kibana-pull-request/builds/292551) failed 2ab64914ef31231cd6df8f0156cf54827e51a32d

cc @ashokaditya

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293399","number":293399,"commit":"d6254594e6cebe8ea27af0226746b46e061c90b0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293312","number":293312,"commit":"84412b660f1976eae18f2a481110455395249cfd"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293251","number":293251,"commit":"64099c7e6f65d0beef03fd943b35d85b5098b86f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292831","number":292831,"commit":"020c0224edc3039635de7b2d61e7c6fb1ed4a695"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292596","number":292596,"commit":"416a928e3f440f3c2d44aae4ceb7a4864ca516ad"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292551","number":292551,"commit":"2ab64914ef31231cd6df8f0156cf54827e51a32d"}],"number":293399}
buildkite-pr-comment-->

ashokaditya committed:
Merge branch 'main' into task/ai4soc-hide-security-sub-privileges

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/293529)
* Commit: 448d39f3fa545789293b53f5556e042c1e7bb908

### Failed CI Steps
* [FTR Configs #41](https://buildkite.com/elastic/kibana-pull-request/builds/293529#01963d41-af28-4d3b-9686-5d86d93176aa)
* [FTR Configs #41](https://buildkite.com/elastic/kibana-pull-request/builds/293529#01963d6e-5d4b-45a8-a5b4-84e583ea3b91)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/293529#01963d41-af28-4d3b-9686-5d86d93176aa) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/293529/jobs/01963d41-af28-4d3b-9686-5d86d93176aa/artifacts/01963d6d-eced-4130-bdb5-7aa4e743de9a) FTR Configs #<span></span>41 / security APIs - Login Selector Login Selector SAML should be able to have many pending SP initiated logins all successfully succeed
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/293529#01963d6e-5d4b-45a8-a5b4-84e583ea3b91) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/293529/jobs/01963d6e-5d4b-45a8-a5b4-84e583ea3b91/artifacts/01963d74-4076-4a73-9e32-8a0593c03a62) FTR Configs #<span></span>41 / security APIs - Login Selector Login Selector SAML should be able to have many pending SP initiated logins all successfully succeed

### History
* :broken_heart: [Build #293399](https://buildkite.com/elastic/kibana-pull-request/builds/293399) failed d6254594e6cebe8ea27af0226746b46e061c90b0
* :broken_heart: [Build #293312](https://buildkite.com/elastic/kibana-pull-request/builds/293312) failed 84412b660f1976eae18f2a481110455395249cfd
* :broken_heart: [Build #293251](https://buildkite.com/elastic/kibana-pull-request/builds/293251) failed 64099c7e6f65d0beef03fd943b35d85b5098b86f
* :green_heart: [Build #292831](https://buildkite.com/elastic/kibana-pull-request/builds/292831) succeeded 020c0224edc3039635de7b2d61e7c6fb1ed4a695
* :broken_heart: [Build #292596](https://buildkite.com/elastic/kibana-pull-request/builds/292596) failed 416a928e3f440f3c2d44aae4ceb7a4864ca516ad

cc @ashokaditya

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293529","number":293529,"commit":"448d39f3fa545789293b53f5556e042c1e7bb908"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293399","number":293399,"commit":"d6254594e6cebe8ea27af0226746b46e061c90b0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293312","number":293312,"commit":"84412b660f1976eae18f2a481110455395249cfd"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293251","number":293251,"commit":"64099c7e6f65d0beef03fd943b35d85b5098b86f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292831","number":292831,"commit":"020c0224edc3039635de7b2d61e7c6fb1ed4a695"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/292596","number":292596,"commit":"416a928e3f440f3c2d44aae4ceb7a4864ca516ad"}],"number":293529}
buildkite-pr-comment-->

ashokaditya committed:
Merge branch 'main' into task/ai4soc-hide-security-sub-privileges

ashokaditya committed:
Merge branch 'main' into task/ai4soc-hide-security-sub-privileges

ashokaditya committed:
Merge branch 'main' into task/ai4soc-hide-security-sub-privileges

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/293793)
* Commit: ef3c9fb2685e8503d140e2546925054b09431d6d

### Failed CI Steps
* [Jest Integration Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/293793#01963f10-8c92-4d10-a603-e33985619acc)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/f00f83715c8787033c7904c0794350f847900bfe) | [after](https://github.com/elastic/kibana/commit/ef3c9fb2685e8503d140e2546925054b09431d6d) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 89.2KB | 89.2KB | +50.0B |
| `securitySolutionEss` | 33.5KB | 33.6KB | +50.0B |
| `securitySolutionServerless` | 48.3KB | 48.3KB | +34.0B |
| total |  |  | +134.0B |



### History
* :yellow_heart: [Build #293632](https://buildkite.com/elastic/kibana-pull-request/builds/293632) was flaky 09e10a26f9bd11b8128554937a39b8f7150d94d8
* :broken_heart: [Build #293551](https://buildkite.com/elastic/kibana-pull-request/builds/293551) failed a8d0ff0a5319cf33e5bdd615f0d367ce0d54342a
* :broken_heart: [Build #293529](https://buildkite.com/elastic/kibana-pull-request/builds/293529) failed 448d39f3fa545789293b53f5556e042c1e7bb908
* :broken_heart: [Build #293399](https://buildkite.com/elastic/kibana-pull-request/builds/293399) failed d6254594e6cebe8ea27af0226746b46e061c90b0
* :broken_heart: [Build #293312](https://buildkite.com/elastic/kibana-pull-request/builds/293312) failed 84412b660f1976eae18f2a481110455395249cfd
* :broken_heart: [Build #293251](https://buildkite.com/elastic/kibana-pull-request/builds/293251) failed 64099c7e6f65d0beef03fd943b35d85b5098b86f

cc @ashokaditya

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293793","number":293793,"commit":"ef3c9fb2685e8503d140e2546925054b09431d6d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293632","number":293632,"commit":"09e10a26f9bd11b8128554937a39b8f7150d94d8"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293551","number":293551,"commit":"a8d0ff0a5319cf33e5bdd615f0d367ce0d54342a"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293529","number":293529,"commit":"448d39f3fa545789293b53f5556e042c1e7bb908"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293399","number":293399,"commit":"d6254594e6cebe8ea27af0226746b46e061c90b0"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/293312","number":293312,"commit":"84412b660f1976eae18f2a481110455395249cfd"}],"number":293793}
buildkite-pr-comment-->

kibanamachine commented:
Starting backport for target branches: 8.19

https://github.com/elastic/kibana/actions/runs/15342953499

kibanamachine commented:
## 💔 All backports failed

| Status | Branch | Result |
|:------:|:------:|:------|
|❌|8.19|**Backport failed because of merge conflicts**<br><br>You might need to backport the following PRs to 8.19:<br> - [[Fleet] Introduce new config setting `xpack.fleet.agentless.isDefault` to set agentless deployment by default (#216535)](https://github.com/elastic/kibana/pull/216535)|

### Manual backport
To create the backport manually run:
```
node scripts/backport --pr 217210
```

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 217210` locally
cc: @ashokaditya

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 217210` locally
cc: @ashokaditya

ashokaditya commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.19|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/222745">](https://github.com/elastic/kibana/pull/222745)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 10.0.0 -->

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 217210` locally
cc: @ashokaditya

ashokaditya commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.19|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/222933">](https://github.com/elastic/kibana/pull/222933)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 10.0.0 -->
